torch_xla/csrc/elementwise.cpp

```diff
@@ -188,6 +188,11 @@ xla::XlaOp BuildReciprocal(xla::XlaOp input) {
   return xla::Div(one, input);
 }
 
+xla::XlaOp BuildSgn(xla::XlaOp input) {
+  xla::XlaOp num_input = ConvertToNumeric(input);
+  return xla::Sign(num_input);
```

Comment:
For the most part I think `xla::Sign` should do what `torch.sgn` do but I would doule check the behavior when the input is `nan`, `0` or `inf`/`-inf`. One easy way to do that is
```
>>> x = torch.randn(2,2, dtype=torch.cfloat)
>>> x[0][0]
tensor(-0.5646+0.1704j)
>>> x[0][0].real.div_(0)
tensor(-inf)
>>> x[0][0]
tensor(-inf+0.1704j)
```

I think pytorch should have test for these(you should check), if not, we can add one to `test_operations`.

Reply:
Please review my initial draft of this test.